### PR TITLE
fixed lvg idempotency issue for all actions: create/extendvg/modify/r…

### PR DESCRIPTION
**Pull request by jdejoya17**
- make sure the following `lvg` actions are idempotent: (1) creating a volume group; (2) extending an existing volume group; (3) modifying an existing volume group; (4) reducing the number of physical volumes in an existing volume group; (5) removing a volume group; (6) varying off/deactivating a volume group; and (7) varying on/activating a volume group
- updated .gitignore

### CODE REVIEW

File: plugins/modules/lvg.py [hunk 1]
```diff
@@ -525,103 +526,213 @@ def vary_vg(module, state, vg_name, vg_state):
...
+    if not just_pvs:
+        cmd = "lsvg -P %s" % vg_name
+        fail_msg = "Failed to fetch information regarding mirror pools of volume \
+            group %s" % vg_name
+        run_cmd(module, cmd, None, fail_msg, fetch=True)
+        init_props += result['stdout']
+
+    return init_props
+
+
```
Comment (pvtorres) [inline plugins/modules/lvg.py:618]: Can you move this function up in such way it is easier for the diff tool to compare side by side?

File: plugins/modules/lvg.py [hunk 2]
```diff
@@ -47,7 +47,7 @@
     - Specifies the type of the volume group.
     - Can be used when creating or changing a volume group, hence when I(state=present).
     type: str
-    choices: [ big, scalable, none ]
+    choices: [ small, big, scalable, None ]
```
Comment (pvtorres) [inline plugins/modules/lvg.py]: Why did you add the small option?

File: plugins/modules/lvg.py [hunk 3]
```diff
@@ -194,16 +194,46 @@
...
+    pvs: hdisk1
+
+- name: Remove a volume group
+  ibm.power_aix.lvg:
+    state: absent
+    vg_name: datavg
+
+- name: Removing hdisk1 and hdisk2 physical volumes from a volume group
+  ibm.power_aix.lvg:
+    state: abset
```
Comment (pvtorres) [inline plugins/modules/lvg.py]: abset to absent

Comment (jdejoya17) [reply plugins/modules/lvg.py]: okay, i will fix that typo

File: plugins/modules/lvg.py [hunk 4]
```diff
@@ -418,89 +369,40 @@ def make_vg(module, vg_name, vg_state):
...
     if vg_type:
-        cmd += vg_type_opt[vg_type]
-
-    enhanced_con_vg = module.params["enhanced_concurrent_vg"]
-    if enhanced_con_vg:
-        cmd += '-C '
-
-    critical_pvs = module.params["critical_pvs"]
-    if critical_pvs:
-        cmd += '-e y '
```
Comment (pvtorres) [inline plugins/modules/lvg.py:440]: At the build_vg_opts (right), if the critical_pvs is false, the cmd+=' -e n" is added. I don't see this in the older code. If that still ok?

File: plugins/modules/lvg.py [hunk 5]
```diff
@@ -418,89 +369,40 @@ def make_vg(module, vg_name, vg_state):
...
-        cmd += "-p %s " % mirror_pool
-
-    num_partitions = module.params["num_partitions"]
-    if num_partitions:
-        cmd += "-P %s " % num_partitions
-
-    critical_vg = module.params["critical_vg"]
-    if critical_vg:
-        cmd += "-r y "
+        opt += "-p %s " % mirror_pool
```
Comment (pvtorres) [inline plugins/modules/lvg.py:311]: Same comment as before. In the build_vg_opts, the "-r n" will be added for non critical_vg. Is that OK?

File: plugins/modules/lvg.py (hunk 4 above)
Comment (pvtorres) [reply plugins/modules/lvg.py:440]: Answer from Joseph. It is not needed for make vg because the default is " -e n". The refactor function, in case the user adds the critical_pvs as false, it will add the '-e n' and that will actually be ignored by the command as it is the default.

File: plugins/modules/lvg.py [hunk 6]
```diff
@@ -255,161 +287,80 @@ def find_vg_state(module, vg_name):
...
+    mirror_pool = module.params["mirror_pool"]
+    if major_num or pp_size or mirror_pool:
+        result['msg'] += "Attributes major_num, pp_size or mirror_pool "
+        result['msg'] += "are not supported while changing volume group %s.\n" % vg_name
+
+    # get initial vg type
+    pattern = r"^(MAX PPs per VG:)\s+(\d+)"
+    max_pp_per_vg = re.search(pattern, init_props, re.MULTILINE)
+    max_pp_per_vg = int(max_pp_per_vg.groups()[1])
+    if max_pp_per_vg == 32512:
```
Comment (raeyacld) [inline plugins/modules/lvg.py]: Can we define the variables for MAX PP size for small,big and scalable VG here?

File: plugins/modules/lvg.py [hunk 7]
```diff
@@ -418,89 +369,40 @@ def make_vg(module, vg_name, vg_state):
     global result
 
     pvs = module.params['pvs']
-
-    cmd = "mkvg "
+    opt = build_vg_opts(module)
 
     vg_type_opt = {
```
Comment (raeyacld) [inline plugins/modules/lvg.py:374]: This is not consistent, are we supposed to add small here? I don't see "small" in the dict.

Comment (raeyacld) [inline plugins/modules/lvg.py:374]: This is not consistent, are we supposed to add small here? I don't see "small" added into the dict.

File: plugins/modules/lvg.py [hunk 8]
```diff
@@ -632,7 +743,7 @@ def main():
         argument_spec=dict(
             state=dict(type='str', choices=['absent', 'present', 'varyoff', 'varyon'], default='present'),
             vg_name=dict(type='str', required=True),
-            vg_type=dict(type='str', choices=['none', 'big', 'scalable']),
+            vg_type=dict(type='str', choices=['small', 'big', 'scalable', None]),
```
Comment (raeyacld) [inline plugins/modules/lvg.py]: so "small" is added to the default VG type in vg_type dict, but I don't see "small" is handled at line 374 vg_type_opt. are we doing this on purpose?

File: plugins/modules/lvg.py [hunk 9]
```diff
@@ -418,89 +369,40 @@ def make_vg(module, vg_name, vg_state):
...
     pvs = module.params['pvs']
-
-    cmd = "mkvg "
+    opt = build_vg_opts(module)
 
     vg_type_opt = {
         "none": '',
         "big": '-B ',
         "scalable": '-S ',
     }
```
Comment (raeyacld) [inline plugins/modules/lvg.py:299]: Should we add small here? The "small" vg type was added to main argument_spec.

File: plugins/modules/lvg.py (hunk 8 above)
Comment (raeyacld) [reply plugins/modules/lvg.py]: If we decide to add small to vg type option, we need to update the documentation at line 50

**Review by JLRebello — APPROVED** [review]

File: plugins/modules/lvg.py (hunk 6 above)
Comment (jdejoya17) [reply plugins/modules/lvg.py]: Ah yes, that would look cleaner, i will be doing that. define it as constants in this module

File: plugins/modules/lvg.py (hunk 9 above)
Comment (jdejoya17) [reply plugins/modules/lvg.py:299]: no, we should not be adding small here. however you are right. i forgot to cleanup the argument spec. i will be removing 'small' in there. thanks

File: plugins/modules/lvg.py (hunk 8 above)
Comment (jdejoya17) [reply plugins/modules/lvg.py]: i will be removing 'small' in here. it is some leftover, that i missed to cleanup during the review we did on the video. thanks for picking it up

**Review by pvtorres — APPROVED** [review]

**Review by nitismis — APPROVED** [review]
